### ISSUE TITLE
Unsupervised Data Augmentation

### ISSUE THREAD

**Issue by agatan** [agatan/paper-reading-notes#21]
- https://arxiv.org/abs/1904.12848

label なしのデータに対して Data Augmentation を適用し、augment する前後での出力分布が一致するように学習することで、大量のラベルなしデータを活かす話。
BERT や RotNet などの事前学習と併用できて、結構な精度向上が確認できている。

### Framework

![image](https://user-images.githubusercontent.com/6925889/57750865-d94df580-771e-11e9-9f28-5b0d3154ac3e.png)

Consistency Loss には KL divergence loss を使っている。

### TSA (Training Signal Anealing)

すこししか labeled data がない場合モデルが簡単に overfit するので、threshold η_t 以上の confidence で正解ラベルを予測できている labeled data については損失に計上しない。
η_t は `1  / (クラス数)` からはじめて、学習が進むに連れて 1 に近づくよう schedule する。

#### Text Classification

Data Augmentation として back-translation を使っている

![image](https://user-images.githubusercontent.com/6925889/57750730-55940900-771e-11e9-9770-5d68aa4f9fd6.png)

- BERT (FINETUNE) ですら UDA を適用すると大きな精度向上が確認できる。 = ターゲットドメインでの事前学習があったとしても UDA 使ったほうが良かった。
- binary classification tasks は full supervised な SOTA に勝っている（！）
    - Pre-BERT SOTA という表現すき

#### Image Classification

- いろんな self-training methods と比較したもの。Architecture は WideResNet-28-2 固定。

![image](https://user-images.githubusercontent.com/6925889/57751229-ff27ca00-771f-11e9-839b-f35510e1b3e2.png)

- 十分な量の supervision があったとして、additional な unlabeled data がどれくらい嬉しいかをはかったもの

![image](https://user-images.githubusercontent.com/6925889/57751447-bcb2bd00-7720-11e9-95b5-b3186043f32c.png)
